### PR TITLE
feat: Add VoicesAppBar

### LINKED ISSUE #491: 🛠️ [TASK] : Header Component
### Summary

Header component in flutter application

### Description

Development of Header component according to https://github.com/input-output-hk/catalyst-voices/issues/380 and the Figma file.

### PR DESCRIPTION
**Pull request by coire1**
# Description

This PR includes:
- a `VoicesAppBar` Widget that can be used as an AppBar in a Scaffold. It follows the UI design for mobile, tablet and desktop.
- a set of Widgets that can be used as actions in the `VoicesAppBar` (they reflect the UI design).
- the CatalystVoicesIcons set to load the custom icon set used in the UI design.

## Related Issue(s)

Closes #491 

## Screenshots

flutter_01
flutter_02
flutter_03
flutter_04
flutter_05

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [ ] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: .config/dictionaries/project.dic [hunk 1]
```diff
@@ -23,6 +23,7 @@ CBOR
 cborg
 CEST
 cfbundle
+cheveron
```
Comment (dtscalac) [inline .config/dictionaries/project.dic:26]: Shouldn't it be "chevron"?

File: catalyst_voices/lib/pages/widgets/voices_app_bar/actions/finish_account_button.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+/// A [FinishAccountButton] widget that is used to display a call to action to
+/// complete a pending account registration.
+class FinishAccountButton extends StatelessWidget {
+  final void Function()? onPressed;
```
Comment (dtscalac) [inline catalyst_voices/lib/pages/widgets/voices_app_bar/actions/finish_account_button.dart:7]: `final VoidCallback? onPressed;`

Comment (dtscalac) [reply catalyst_voices/lib/pages/widgets/voices_app_bar/actions/finish_account_button.dart:7]: make sure to apply in other places, fyi: another popular defined typedef is `ValueChanged<T>`

File: .config/dictionaries/project.dic (hunk 1 above)
Comment (coire1) [reply .config/dictionaries/project.dic:26]: Not according to he UI kit icons :) Let me check on Figma and change them there!

**Review by dtscalac — APPROVED** [review]

**Review by dtscalac — APPROVED** [review]

**Pull request merged**